### PR TITLE
use pengutronix-hosted runner

### PR DESCRIPTION
**Pull request by jluebbe**
This is based on @hnez's [forrest runner](https://github.com/hnez/forrest), which provides ephemeral VMs while still allowing efficient and secure access to download and sstate caches.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: build
+
+on:
+  push: {}
+  pull_request: {}
+  # allow rebuilding without a push
+  workflow_dispatch: {}
```
Comment (hnez) [inline .github/workflows/build.yml:7]: I would suggest adding a weekly or monthly [`on: schedule:`](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule) run to keep the sstate fresh and to check if everything still works with the current master of all the other layers.

But that's an unrelated change that could be done in a separate PR.

**Review by hnez — APPROVED** [review]

**Review by ejoerns — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    name: meta-rauc Build
+    # run on self-hosted runner for the main repo or if vars.BUILD_RUNS_ON is set
+    runs-on: >-
+      ${{
+        (vars.BUILD_RUNS_ON != ''  && fromJSON(vars.BUILD_RUNS_ON)) ||
+        (github.repository == 'rauc/meta-rauc' && fromJSON('["self-hosted", "forrest", "build"]')) ||
+        'ubuntu-20.04'
+      }}
+    # abort if it seems that we're rebuilding too much
+    timeout-minutes: 120
```
Comment (ejoerns) [inline .github/workflows/build.yml:20]: Won't we get stuck here then? Currently, we don't save the sstate cache on errors/timeouts, right?

Comment (jluebbe) [reply .github/workflows/build.yml:20]: A full rebuild on our runner needs less than 2 hours.

Comment (ejoerns) [reply .github/workflows/build.yml:20]: Sounds promising.

**Pull request merged**